scipy/stats/tests/test_resampling.py

```diff
@@ -1683,20 +1693,21 @@ def test_against_binomtest(self, alternative):
         # So we have to work with +1/-1 instead of 1/0.
 
         def statistic(x, axis=0):
-            return np.sum(x > 0, axis=axis)
+            xp_ = array_namespace(x)
+            return xp_.count_nonzero(x > 0, axis=axis)
 
         k, n, p = statistic(x), 10, 0.5
         expected = stats.binomtest(k, n, p, alternative=alternative)
 
-        res = stats.permutation_test((x,), statistic, vectorized=True,
-                                     permutation_type='samples',
-                                     n_resamples=np.inf, rng=self.rng,
-                                     alternative=alternative)
-        assert_allclose(res.pvalue, expected.pvalue, rtol=self.rtol)
+        res = stats.permutation_test((xp.asarray(x, dtype=xp.float64),), statistic,
+                                     permutation_type='samples', n_resamples=np.inf,
```

Comment:
looks like this passes with `np.inf` -> `xp.inf`; is that change just pointless or?

Reply:
Switched to xp.inf.